### ISSUE TITLE
Sentry problems in XCode project

### ISSUE THREAD

**Issue by perholmes** [getsentry/sentry-unity#1343]
We're having what might be the same problem as issue ###, that Sentry has significant issues that prevent building an XCode project. This is both Sentry 1.2.0 and 1.3.0, using Unity 2023.1.0a25 and XCode 14.

We have found a workaround, but we don't know why it works.

Basically, whether or not iOS Native Support is enabled, XCode producers a bunch of link errors around SentryNativeBridge, such as linker errors for "_SentryNativeBridgeAddBreadcrumb". This is coming from function in the Sentry.xcframework folder not being compiled.

Curiously, the Unity platform settings for SentryNativeBridge do not enable iOS. So we've had to embed Sentry into the Unity project to get to enable this. This gets us half of the way.

But more curiously, Sentry.xcframework~ has a tilde at the end, prevent everything in that folder from being visible to Unity. Removing the tilde, and instead also enabling iOS for Sentry.xcframework/ios-arm64/Sentry and Sentry.xcframework/Sentry.xcframework/ios-arm64/Sentry, it's now possible to complete an XCode build.

Clearly, whatever is in Sentry.xcframework~ is very very important, and shouldn't be made hidden. But we can't see through why Sentry is designed this way. Do you have another way of enabling it for the XCode build that might be failing? This isn't a problem for Android builds.

That's what we know, but I would be great to understand what's going on here, and whether we're simply using it wrong.

**Comment by perholmes** [comment]
Nope, didn't work when deployed on device. Some Sentry module missing.

I found that going back to 1.1.0, Sentry builds and works. So whatever broke is between 1.1.0 and 1.3.0. Version 1.2.0 worked on Android, Windows and Mac, so it has to be something iOS-specific that broke.

**Comment by bitsandfoxes** [comment]
Thanks for the detailed report!
We have not looked into `Unity 2023` so far as it's still a pre-release but it's a good call to stay ahead of this.
I'll try to reproduce this issue locally and then update here.

**Comment by perholmes** [comment]
A data point to keep in mind is that Sentry Unity 1.1.0 builds fine in the same Unity alpha. So I'm not 100% sure this is attributable to this being a pre-release Unity alpha.

I'm happy to provide a build log, but they're immense, and also a bit privileged, so I'd rather not post them in a public forum. Is there somewhere else I can send it?

**Comment by bitsandfoxes** [comment]
>I'm happy to provide a build log, but they're immense, and also a bit privileged, so I'd rather not post them in a public forum. Is there somewhere else I can send it?

Those would be highly appreciated. stefan@sentry.io

**Comment by bitsandfoxes** [comment]
Apologies, this issue has not been properly linked to it's PR: https://github.com/getsentry/sentry-unity/pull/1352 has a fix has been release with `1.4.1`.
